File: lib/src/core/application_common_providers.dart
library angular2.src.core.application_common_providers;

import "package:angular2/src/facade/lang.dart" show Type;
import "package:angular2/src/core/di.dart"
    show provide, Provider, Injector, OpaqueToken;
import "application_tokens.dart"
    show APP_COMPONENT_REF_PROMISE, APP_COMPONENT, APP_ID_RANDOM_PROVIDER;
import "change_detection/change_detection.dart"
    show
        IterableDiffers,
        defaultIterableDiffers,
        KeyValueDiffers,
        defaultKeyValueDiffers;
import "linker/view_pool.dart" show AppViewPool, APP_VIEW_POOL_CAPACITY;
import "linker/view_manager.dart" show AppViewManager;
import "linker/view_manager.dart" show AppViewManager_;
import "linker/view_manager_utils.dart" show AppViewManagerUtils;
import "linker/view_resolver.dart" show ViewResolver;
import "linker/view_listener.dart" show AppViewListener;
import "linker/proto_view_factory.dart" show ProtoViewFactory;
import "linker/directive_resolver.dart" show DirectiveResolver;
import "linker/pipe_resolver.dart" show PipeResolver;
import "linker/compiler.dart" show Compiler;
import "linker/compiler.dart" show Compiler_;
import "linker/dynamic_component_loader.dart" show DynamicComponentLoader;
import "linker/dynamic_component_loader.dart" show DynamicComponentLoader_;
import "render.dart" show EventManager;

/**
 * A default set of providers which should be included in any Angular
 * application, regardless of the platform it runs onto.
 */
const List<
        dynamic /* Type | Provider | List < dynamic > */ > APPLICATION_COMMON_PROVIDERS =
    const [
  const Provider(Compiler, useClass: Compiler_),
  APP_ID_RANDOM_PROVIDER,
  AppViewPool,
  const Provider(APP_VIEW_POOL_CAPACITY, useValue: 10000),
  const Provider(AppViewManager, useClass: AppViewManager_),
  AppViewManagerUtils,
  AppViewListener,
  ProtoViewFactory,
  ViewResolver,
  const Provider(IterableDiffers, useValue: defaultIterableDiffers),
  const Provider(KeyValueDiffers, useValue: defaultKeyValueDiffers),
  DirectiveResolver,
  PipeResolver,
  const Provider(DynamicComponentLoader, useClass: DynamicComponentLoader_),
  EventManager
];

File: ts/src/core/application_common_providers.ts
import {Type, CONST_EXPR} from 'angular2/src/facade/lang';
import {provide, Provider, Injector, OpaqueToken} from 'angular2/src/core/di';
import {
  APP_COMPONENT_REF_PROMISE,
  APP_COMPONENT,
  APP_ID_RANDOM_PROVIDER
} from './application_tokens';
import {
  IterableDiffers,
  defaultIterableDiffers,
  KeyValueDiffers,
  defaultKeyValueDiffers
} from './change_detection/change_detection';
import {ResolvedMetadataCache} from 'angular2/src/core/linker/resolved_metadata_cache';
import {AppViewManager} from './linker/view_manager';
import {AppViewManager_} from "./linker/view_manager";
import {ViewResolver} from './linker/view_resolver';
import {AppViewListener} from './linker/view_listener';
import {DirectiveResolver} from './linker/directive_resolver';
import {PipeResolver} from './linker/pipe_resolver';
import {Compiler} from './linker/compiler';
import {Compiler_} from "./linker/compiler";
import {DynamicComponentLoader} from './linker/dynamic_component_loader';
import {DynamicComponentLoader_} from "./linker/dynamic_component_loader";

/**
 * A default set of providers which should be included in any Angular
 * application, regardless of the platform it runs onto.
 */
export const APPLICATION_COMMON_PROVIDERS: Array<Type | Provider | any[]> = CONST_EXPR([
  new Provider(Compiler, {useClass: Compiler_}),
  APP_ID_RANDOM_PROVIDER,
  ResolvedMetadataCache,
  new Provider(AppViewManager, {useClass: AppViewManager_}),
  AppViewListener,
  ViewResolver,
  new Provider(IterableDiffers, {useValue: defaultIterableDiffers}),
  new Provider(KeyValueDiffers, {useValue: defaultKeyValueDiffers}),
  DirectiveResolver,
  PipeResolver,
  new Provider(DynamicComponentLoader, {useClass: DynamicComponentLoader_})
]);